sktime/forecasting/base/adapters/_statsmodels.py

```diff
@@ -111,12 +111,15 @@ def _predict(self, fh, X):
         # statsmodels exog must contain test indices only.
         # For discussion see https://github.com/sktime/sktime/issues/3830
         if X is not None:
-            ind_drop = self._X.index
-            X = X.loc[~X.index.isin(ind_drop)]
+            if self._X is not None:
```

Comment:
Optional: may be merge with previous check with `and`?

Reply:
I will check this out.

Reply:
Can be done ,I'll make the change.